apps/ewallet_db/lib/ewallet_db/account.ex

```diff
@@ -97,28 +98,23 @@ defmodule EWalletDB.Account do
     |> put_categories(attrs, :category_ids)
   end
 
-  # Since the attribute key could be either an atom or string, the two functions below
-  # are created to support both types.
-  defp put_categories(changeset, attrs, attr_name) when is_atom(attr_name) do
-    category_ids = Map.get(attrs, attr_name) || Map.get(attrs, to_string(attr_name))
-    put_categories(changeset, category_ids)
-  end
+  defp put_categories(changeset, attrs, attr_name) do
+    case InputAttribute.get(attrs, attr_name) do
+      ids when is_list(ids) ->
+        put_categories(changeset, ids)
 
-  defp put_categories(changeset, attrs, attr_name) when is_binary(attr_name) do
-    category_ids = Map.get(attrs, attr_name) || Map.get(attrs, String.to_existing_atom(attr_name))
-    put_categories(changeset, category_ids)
+      _ ->
+        changeset
+    end
   end
 
-  defp put_categories(changeset, category_ids) when is_list(category_ids) do
+  defp put_categories(changeset, category_ids) do
     # Associations need to be preloaded before updating
     changeset = Map.put(changeset, :data, Repo.preload(changeset.data, :categories))
     categories = Repo.all(from(c in Category, where: c.id in ^category_ids))
```

Comment:
I think we could use the `UUIDFetcher` for this, to make it more reusable throughout controllers. Good enough for now however, since this is a peculiar case. We might want to update it later :)